JS/Lesson17/sinan/script.js

```diff
@@ -134,38 +134,84 @@ console.log('----------------------------');
 // Input: ({name: "John", age: 30}, "city", "New York")
 // Output: {name: "John", age: 30, city: "New York"}
 
+function getKeyValue (obj,key,value) {
+if(obj[key]=value) return obj;
+} 
+
+const newValue = getKeyValue({name: "John", age: 30}, "city", "New York");
+console.log('Exercise 10:',newValue);
+
 console.log('----------------------------');
 // 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
 // Input: ({name: "John", age: 30, city: "New York"}, "city")
 // Output: {name: "John", age: 30}
 
+function removeKeyValue(testObject, key) {
+  if (testObject.hasOwnProperty(key)) {
+    delete testObject[key];
+    return testObject;
+  } 
+}
+  const ourList = removeKeyValue({name: "John", age: 30, city: "New York"}, "city");
+  console.log('Exercise 11:',ourList);
+
 console.log('----------------------------');
 // 12. Write a function that takes two arrays of equal length and returns an object with the first array as keys and the second array as values.
 // Input: (["name", "age", "city"], ["John", 30, "New York"])
 // Output: {name: "John", age: 30, city: "New York"}
+function concatKeyValue(list1,list2,) {
+  const unifiedList = [];
```

Comment:
What you need to do in this exercise is to create a new object from the two arrays:
```
if (list1.length === list2.length) {
  let newObject = {}; // Create a new empty object first
  for (let i = 0; i < list1.length; i++) {
    newObject[list1[i]] = list2[i];
  }
  return newObject;
} else  console.error('provided arrays are not of equal length!');

```